### DISCUSSION TITLE
In single stage pipeline, how memory loading from shared ram is overlapped with mma computation?

### DISCUSSION THREAD

**Discussion by wzhcz8902** [NVIDIA/cutlass#1503] [Q&A] [answered]
https://github.com/NVIDIA/cutlass/blob/5c447dd84f8ae0e1d48ff9a2eae26ce8c4958101/include/cutlass/gemm/threadblock/mma_singlestage.h#L198-L201
As the comment say there is some overlap between memory loading from shared ram with mma, but in the loop
https://github.com/NVIDIA/cutlass/blob/5c447dd84f8ae0e1d48ff9a2eae26ce8c4958101/include/cutlass/gemm/threadblock/mma_singlestage.h#L223-L239
I do not see such behavior, can you help me understand it?

**Comment by hwu36** [comment, marked as answer]
There is no overlap in single stage.

just

```
loop:
  load
  compute
```

It saves space at the cost of efficiency.
